### DISCUSSION TITLE
Any plans on using TypeScript?

### DISCUSSION THREAD

**Discussion by JeroenReumkens** [nl-design-system/backlog#16] [Feedback gevraagd] [answered]
I was wondering whether there are any plans on using TypeScript? I think this can save you a lot of headaches in the long run 😁 

I can always help you with the initial setup.

**Comment by Robbert** [comment, marked as answer]
We do currently expect it will be essential for adoption in certain stacks to provide TypeScript definitions for the components in their npm packages, so using TypeScript to that extent is certain.

Using TypeScript for the implementation of all components is not on our radar yet, but I think contributed components in TypeScript could very well exist alongside ES Module components.

Using TypeScript also slightly complicates the setup of testing with Jest, if I recall correctly.

Once we have the first TypeScript contribution, we would need to make an effort to make the `tsconfig.json` scale to many packages and have a very strict configuration right off the bat (`"noImplicitAny": true`).
